lib/kvdb/kvdb_ctxn.c

```diff
@@ -87,6 +101,7 @@ kvdb_ctxn_trylock_impl(struct kvdb_ctxn_impl *ctxn)
 
     return 0;
 }
+// kvdb_ctxn_trylock_impl() is a misleading name. How about kvdb_ctxn_lock_if_active()?
```

Comment:
If you change the name then you need to change the name of kvdb_ctxn_trylock_write() to kvdb_ctxn_lock_if_active_and_can_begin_and_can_enable_inserts_and_can_pfxlock_and_can_keylock();

I think maybe it's misleading to someone who hasn't bothered to look at the implementation and sees "trylock" and thinks just because they see trylock they have some immediate and complete understanding of the function..
It's "kvdb_ctxn_trylock" and not "pthread_mutex_trylock()", and while they have similar semantics at a high level one shouldn't infer that they have the exact same semantics...

Reply:
99% of the time trylock means "try to take the lock w/o waiting".  But this one waits.

Reply:
It used to do a trylock.  I think we changed it a while back to give the illusion that the API was thread safe..

Reply:
FWIW, it's still a trylock in the sense that it can effectively fail to obtain the lock.